File: fixtures/java/baseline/src/main/java/example/Service.java
package example;

public class Service {
    public static final String DEFAULT_PREFIX = "job";

    private final Repository repository;

    public Service(Repository repository) {
        this.repository = repository;
    }

    public String execute(String name) {
        String stored = repository.save(name);
        return DEFAULT_PREFIX + ":" + stored;
    }

    public static class Repository {
        public String save(String value) {
            return value.trim();
        }
    }
}

File: fixtures/go/baseline/src/example/service.go
package example

type Repository interface {
    Save(value string) string
}

const DefaultPrefix = "job"

var DefaultRepository Repository = &MemoryRepository{}

type MemoryRepository struct {
    Last string
}

func (m *MemoryRepository) Save(value string) string {
    m.Last = value
    return value
}

type Service struct {
    repository Repository
}

func NewService(repository Repository) Service {
    return Service{repository: repository}
}

func (s Service) Execute(name string) string {
    stored := s.repository.Save(name)
    return DefaultPrefix + ":" + stored
}
